icssc/peterportal-public-api
# Maintenance tasks

## Reference Issues

N/A

## Summary of Change/Fix

* Commit the new lockfile created by node 16+/npm v7+, so new clones/installs won't print the big scary warning about it being outdated.
Rationale: The Fellowship says to install node v16.13.2 with `nvm`, so it's probably reasonable to expect that most of our new contributors will be using that version (unless we update the README to state otherwise). There's also not really any reason to stick with node 14, since it will be deprecated by April 2023 and Lambda already supports node 16.
* Update our CI/CD pipeline to deploy to node 16.
* Resolve outstanding potential security vulnerabilities as reported by `npm audit` and Dependabot.
* Update some packages to eliminate dependency deprecation as much as possible.
* Remove end-of-life (10.x, 12.x) and non-LTS (15.x) node versions from the tests workflow. We might have unlimited Actions minutes but there's no point in testing on platforms we no longer target.

## Test Plan
* `npm test`
* Verify that docs can still be built properly, since docs_requirements was also patched.
* Verify that the dev Lambda instance is now on node 16.
